driver/makedepend.ml

```diff
@@ -406,6 +407,8 @@ let mli_file_dependencies source_file =
 let process_file_as process_fun def source_file =
   Compenv.readenv ppf (Before_compile source_file);
   load_path := [];
+  if not !nocwd then
+    add_to_load_path Filename.current_dir_name;
```

Comment:
There is a small incompatibility hidden here: the current directory must be added after the addition of `include_dirs` and before the addition of `first_include_dirs`. Indeed with the trunk implementation, if we have a module that can
be resolved to both ./a.ml and path/a.ml, the current working directory takes priority. Typically, if we have 
```
sub/a.ml
a.ml
b.ml
```
and 
```ocaml
(* b.ml *)
open A
```

calling `ocamldep b.ml` and `ocamldep -I sub b.ml` both resolve `A` to `a.ml`.
But with this PR in its current state, `ocamldep -I sub b.ml` resolves `A` to `sub/a.ml` because `.` has the lowest possible priority.

Reply:
That's a really good catch !
would replacing `add_to_load_path Filename.current_dir_name;` with 
`add_to_list first_include_dirs Filename.current_dir_name;` solve this issue ?

Reply:
I think it is better to avoid doing global side-effect here. Just splitting the `List.iter` below  in two to add in order
* `last_include_dirs + include_dirs`
* (conditionally) current_dir_name
* first_include_dir
seems more composable.

Reply:
Does this look better ?